hrms/payroll/doctype/payroll_entry/payroll_entry.py

```diff
@@ -756,138 +839,228 @@ def set_start_end_dates(self):
+
+		for emp in frappe.get_all(
+			"Employee",
+			fields=["name", "date_of_joining", "holiday_list"],
+			filters={"name": ("in", employees)},
+		):
+			employee_detials[emp.name] = {}
+			employee_detials[emp.name].update(
+				{"date_of_joining": emp.date_of_joining, "holiday_list": emp.holiday_list}
 			)
-			if days and days[0][0]:
-				holidays = days[0][0]
-		return holidays
 
-	def get_count_employee_attendance(self, employee, start_date):
-		marked_days = 0
-		attendances = frappe.get_all(
+		return employee_detials
+
+	def get_attendance_for_employees(self, employees):
+		attendance_count = {}
+
+		for att in frappe.get_all(
 			"Attendance",
-			fields=["count(*)"],
-			filters={"employee": employee, "attendance_date": ("between", [start_date, self.end_date])},
-			as_list=1,
+			filters={
+				"employee": ["in", employees],
+				"attendance_date": ["between", [self.start_date, self.end_date]],
+			},
+			fields=["employee", "count(*) as attendance_count"],
+			group_by="employee",
+		):
+
+			attendance_count[att.employee] = {}
+			attendance_count[att.employee]["attendance_count"] = att.attendance_count
+
+		return attendance_count
+
+	def get_holidays_count_for_employee(self, holiday_list, start_date, holidays_count):
+		key = f"{start_date}-{self.end_date}-{holiday_list}"
+
+		if key in holidays_count:
+			return holidays_count[key]
+
+		holidays = frappe.db.get_all(
+			"Holiday",
+			filters={"parent": holiday_list, "holiday_date": ("between", [start_date, self.end_date])},
+			fields=["count(*) as holiday_count"],
+			as_list=True,
 		)
-		if attendances and attendances[0][0]:
-			marked_days = attendances[0][0]
-		return marked_days
+
+		if len(holidays) > 0:
+			holidays_count[key] = holidays[0][0]
+
+		return holidays_count[key] or 0
 
 
 def get_sal_struct(
-	company: str, currency: str, salary_slip_based_on_timesheet: int, condition: str
+	company: str, currency: str, salary_slip_based_on_timesheet: int, payroll_frequency: str
 ):
-	return frappe.db.sql_list(
-		"""
-		select
-			name from `tabSalary Structure`
-		where
-			docstatus = 1 and
-			is_active = 'Yes'
-			and company = %(company)s
-			and currency = %(currency)s and
-			ifnull(salary_slip_based_on_timesheet,0) = %(salary_slip_based_on_timesheet)s
-			{condition}""".format(
-			condition=condition
-		),
-		{
-			"company": company,
-			"currency": currency,
-			"salary_slip_based_on_timesheet": salary_slip_based_on_timesheet,
-		},
+	SalaryStructure = frappe.qb.DocType("Salary Structure")
+	return (
+		frappe.qb.from_(SalaryStructure)
+		.select(SalaryStructure.name)
+		.where(
+			(SalaryStructure.docstatus == 1)
+			& (SalaryStructure.is_active == "Yes")
+			& (SalaryStructure.company == company)
+			& (SalaryStructure.currency == currency)
+			& (SalaryStructure.salary_slip_based_on_timesheet == salary_slip_based_on_timesheet)
+			& (SalaryStructure.payroll_frequency == payroll_frequency)
+		)
+	).run(pluck=True)
+
+
+def get_emp_list(
```

Comment:
This whole get_emp_list function has become extremely confusing now. There is a method called `get_emp_list`, a function `get_employee_list` and one more function `get_emp_list` outside the class.

Commonifying code is okay until it affects readability. This particular scenario is making it more difficult to figure out what's happening. Rethink this whole refactor maybe.